tools/ci_build/ci_sanity.sh

```diff
@@ -0,0 +1,318 @@
+                echo "  ${LINE}"
+                IS_WHITELISTED=1
+            fi
+        done
+
+        if [[ ${IS_WHITELISTED} == "0" ]]; then
+            echo "${LINE}" >> ${NONWL_ERRORS_FILE}
+            echo "" >> ${NONWL_ERRORS_FILE}
+            ((N_ERRORS++))
+        fi
+    done <${ERRORS_FILE}
+
+    echo ""
+    if [[ ${N_ERRORS} != 0 ]]; then
+        echo "FAIL: Found ${N_ERRORS} non-whitelited pylint errors:"
+        cat "${NONWL_ERRORS_FILE}"
+        return 1
+    else
+        echo "PASS: No non-whitelisted pylint errors were found."
+        return 0
+    fi
+}
+
+# Run pep8 check
+do_pep8() {
+    # Usage: do_pep8 [--incremental]
+    # Options:
+    #   --incremental  Performs check on only the python files changed in the
+    #                  last non-merge git commit.
+
+    PEP8_BIN="/usr/local/bin/pep8"
+    PEP8_CONFIG_FILE="${SCRIPT_DIR}/pep8"
+
+    if [[ "$1" == "--incremental" ]]; then
+        PYTHON_SRC_FILES=$(get_py_files_to_check --incremental)
+        NUM_PYTHON_SRC_FILES=$(echo ${PYTHON_SRC_FILES} | wc -w)
+
+        echo "do_pep8 will perform checks on only the ${NUM_PYTHON_SRC_FILES} "\
+             "Python file(s) changed in the last non-merge git commit due to the "\
+             "--incremental flag:"
+        echo "${PYTHON_SRC_FILES}"
+        echo ""
+    else
+        PYTHON_SRC_FILES=$(get_py_files_to_check)
+    fi
+
+    if [[ -z ${PYTHON_SRC_FILES} ]]; then
+        echo "do_pep8 found no Python files to check. Returning."
+        return 0
+    fi
+
+    if [[ ! -f "${PEP8_CONFIG_FILE}" ]]; then
+        die "ERROR: Cannot find pep8 config file at ${PEP8_CONFIG_FILE}"
+    fi
+    echo "See \"${PEP8_CONFIG_FILE}\" for pep8 config( e.g., ignored errors)"
+
+    NUM_SRC_FILES=$(echo ${PYTHON_SRC_FILES} | wc -w)
+
+    echo "Running pep8 on ${NUM_SRC_FILES} files"
+    echo ""
+
+    PEP8_START_TIME=$(date +'%s')
+    PEP8_OUTPUT_FILE="$(mktemp)_pep8_output.log"
+
+    rm -rf ${PEP8_OUTPUT_FILE}
+
+    ${PEP8_BIN} --config="${PEP8_CONFIG_FILE}" --statistics \
+        ${PYTHON_SRC_FILES} 2>&1 | tee ${PEP8_OUTPUT_FILE}
+    PEP8_END_TIME=$(date +'%s')
+
+    echo ""
+    echo "pep8 took $((PEP8_END_TIME - PEP8_START_TIME)) s"
+    echo ""
+
+    if [[ -s ${PEP8_OUTPUT_FILE} ]]; then
+        echo "FAIL: pep8 found above errors and/or warnings."
+        return 1
+    else
+        echo "PASS: No pep8 errors or warnings were found"
+        return 0
+    fi
+}
+
+
+#Check for the bazel cmd status (First arg is error message)
+cmd_status(){
+    if [[ $? != 0 ]]; then
+        echo ""
+        echo "FAIL: ${BUILD_CMD}"
+        echo "  $1 See lines above for details."
+        return 1
+    else
+        echo ""
+        echo "PASS: ${BUILD_CMD}"
+        return 0
+    fi
+}
+
+# Run bazel build --nobuild to test the validity of the BUILD files
+do_bazel_nobuild() {
+    BUILD_TARGET="//tensorflow_addons/..."
+    BUILD_CMD="bazel build --nobuild ${BAZEL_FLAGS} -- ${BUILD_TARGET}"
```

Comment:
@karmel  Hi, the check fails because`'@local_config_tf//` package is missing. Do you know how to fix it?

```
 and referenced by '//tensorflow_addons/custom_ops/text:python/_skip_gram_ops.so'
ERROR: Analysis of target '//tensorflow_addons/custom_ops:custom_ops_py' failed; build aborted: no such package '@local_config_tf//': Traceback (most recent call last):
        File "/addons/tf_dependency/tf_configure.bzl", line 177
                _symlink_genrule_for_dir(repository_ctx, tf_header_dir, "incl...", ...")
        File "/addons/tf_dependency/tf_configure.bzl", line 151, in _symlink_genrule_for_dir
                "\n".join(sorted(_read_dir(repository_ctx,...()))
        File "/addons/tf_dependency/tf_configure.bzl", line 151, in "\n".join
                sorted(_read_dir(repository_ctx, src_di...())
        File "/addons/tf_dependency/tf_configure.bzl", line 151, in sorted
                _read_dir(repository_ctx, src_dir)
        File "/addons/tf_dependency/tf_configure.bzl", line 84, in _read_dir
                _execute(repository_ctx, ["find", src_dir, ..."], ...)
        File "/addons/tf_dependency/tf_configure.bzl", line 51, in _execute
                _fail("\n".join([error_msg.strip() if ... ""]))
        File "/addons/tf_dependency/tf_configure.bzl", line 19, in _fail
                fail(("%sPython Configuration Error:%...)))
Python Configuration Error: Repository command failed
find: `/usr/local/lib/python2.7/dist-packages/tensorflow/include': No such file or directory
```

Reply:
A call to `./configure.sh` before this script in the make-file should fix I believe?

Reply:
Thanks, fixed.